subprojects/base-services/src/main/java/org/gradle/internal/reflect/JavaReflectionUtil.java

```diff
@@ -281,6 +287,28 @@ public static boolean propertyExists(Object target, String propertyName) {
         return true;
     }
 
+    public static <T> T newInstanceOrFallback(String jdk7Type, ClassLoader loader, Class<? extends T> fallbackType) {
+        // Use java 7 APIs, if available
```

Comment:
Some services still need to run on older JDKs (example, executes tests in a forked environment with JDK 5).